### PR TITLE
Implement add and remove features

### PR DESCRIPTION
**Pull request by amentetteh**
Opened for :

Remove all hardcoded items from the tasks array.
Add taskcontroller.js
Implement adding a new task (add a new element to the array).
Implement deleting a task (remove an element from the array).
Implement  editing task descriptions.
Implement  sorting and indexes updating

### CODE REVIEW

File: src/js/modules/controller/taskcontroller.js [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+      TaskController.resetForRemove();
+      const newTask = new Task(task.index, task.description, task.completed);
+      console.log('**********************')
+      console.log(newTask)
+      if(newTask){
+        newTask.setCompleted(checkboxElement.checked);
+        console.log(newTask)
+      }
+
+    }); */
```
Comment (aimalamiri) [inline src/js/modules/controller/taskcontroller.js]: Good job! but it's highly recommended to keep your repository clean. So please make sure to remove it.

File: src/js/modules/controller/taskcontroller.js [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+      const index = parseInt(form.getElementsByTagName('div')[0].innerText, 10);
+      const description = form.description.value;
+      const completed = (form.getElementsByTagName('div')[1].innerText === 'true');
+      const task = new Task(index, description, completed);
+      task.save();
+    });
+
+    deleteIcon.addEventListener('click', () => {
+      TaskController.removeTask(TaskController.taskList, task);
+      // TaskController.removeTaskFromUI(event.target)
```
Comment (aimalamiri) [inline src/js/modules/controller/taskcontroller.js]: Good job! but it's highly recommended to keep your repository clean. So please make sure to remove it.

File: src/js/modules/controller/taskcontroller.js [hunk 3]
```diff
@@ -0,0 +1,157 @@
...
+    const newTask = new Task(task.index, task.description, task.completed);
+    newTask.remove();
+    TaskController.resetDisplayTasks(table);
+  };
+
+  /* static removeTaskFromUI = (item) => {
+    if (item.classList.contains("delete-icon")) {
+      item.parentElement.remove();
+    }
+  }; */
```
Comment (aimalamiri) [inline src/js/modules/controller/taskcontroller.js]: Good job! but it's highly recommended to keep your repository clean. So please make sure to remove it.

**Review by aimalamiri — CHANGES_REQUESTED** [review]
Hi @amentetteh, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

AlmostThereReachGIF

## Highlights

- The application is designed very well and look fantastic ✔️ ⭐
- Good job for using the right GitHub Flow 💯
- Awesome file organization 🗃️
- Commits are descriptive and meaningful 👍
- Linters are properly set up and passing 🟢 👍

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/js/modules/controller/taskcontroller.js (hunk 1 above)
Comment (amentetteh) [reply src/js/modules/controller/taskcontroller.js]: OK. Thanks

Comment (Abenezer-Tilahun) [reply src/js/modules/controller/taskcontroller.js]: Fixed.

File: src/js/modules/controller/taskcontroller.js (hunk 2 above)
Comment (Abenezer-Tilahun) [reply src/js/modules/controller/taskcontroller.js]: Fixed.

File: src/js/modules/controller/taskcontroller.js (hunk 3 above)
Comment (Abenezer-Tilahun) [reply src/js/modules/controller/taskcontroller.js]: Fixed.

**Review by Abenezer-Tilahun — APPROVED** [review]
Hi @amentetteh     
 
Your project is complete  👍 💯   There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights 👏🏻

✔️ No linter errors.
✔️ You used correct Github-Flow.
✔️ Professional README  
✔️ Good Commit Messages
✔️ Implement the CRUD  functionality.


## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**